mongoimport/mongoimport.go

```diff
@@ -233,23 +237,21 @@ func (imp *MongoImport) validateSettings(args []string) error {
 		log.Logvf(log.Info, "using upsert fields: %v", imp.upsertFields)
 	}
 
-	// set the number of decoding workers to use for imports
-	if imp.IngestOptions.NumDecodingWorkers <= 0 {
-		imp.IngestOptions.NumDecodingWorkers = imp.ToolOptions.MaxProcs
-	}
-	log.Logvf(log.DebugLow, "using %v decoding workers", imp.IngestOptions.NumDecodingWorkers)
-
-	// set the number of insertion workers to use for imports
-	if imp.IngestOptions.NumInsertionWorkers <= 0 {
-		imp.IngestOptions.NumInsertionWorkers = 1
-	}
-
-	log.Logvf(log.DebugLow, "using %v insert workers", imp.IngestOptions.NumInsertionWorkers)
-
-	// if --maintainInsertionOrder is set, we can only allow 1 insertion worker
 	if imp.IngestOptions.MaintainInsertionOrder {
+		imp.IngestOptions.StopOnError = true
```

Comment:
There is a chance that it could expand to cover more types of errors at some point, so we might not want to get too specific. It's also nice that it exactly mirrors the command-line arg. That being said, I think `StopOnAllInsertionErrors` or `StopOnAllWriteErrors` are pretty accurate for current behavior. `IgnoreErrorFilter` or `BypassErrorFilter` have a little bit more future-proof to them, though. What do you think?